src/Symfony/Component/Validator/Constraints/NotJsonValidator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ */
+class NotJsonValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof NotJson) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\NotJson');
+        }
+
+        if (!$this->isJson($value)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setCode(NotJson::NOT_JSON_ERROR)
+                ->addViolation();
+        }
+    }
+
+    /**
+     * @param string $string
+     *
+     * @return bool
+     */
+    protected function isJson($string)
+    {
+        json_decode($string);
```

Comment:
Can you check if you can make this compatible with the new json exception throwing in php 7.3? See this RFC for the implementation: https://wiki.php.net/rfc/json_throw_on_error

A bit more of a security/performance concern, you're directly decoding json here. What if the user inputs a very large string? I'm not sure if this needs protection but I'm no expert on this. Will it also impact performance as it's basically decoded twice now?

Could be solved by making a `JsonType` that uses a data transformer.

Reply:
> Will it also impact performance as it's basically decoded twice now?

Why not to add the result in the payload of the constraint ?

```php
$constraint->payload['result']  = json_decode($string);
```

Reply:
@iltar, for the new json exception, I'm not sure if that will be compatible with other PHP versions?

 for security/performance concern ( large sting), I think that we are in the same case as the Serializer Component. 
 https://github.com/symfony/symfony/blob/00e5cd9a1c237e579e6327e9a66c512bf76f292a/src/Symfony/Component/Serializer/Encoder/JsonDecode.php#L75

Reply:
No one can provide a feedback for this? can we resolve this discussion?

Reply:
From the RFC:
> As mentioned earlier, it may be desirable to deprecate the default behaviour eventually. 

As this behavior will not be deprecated just yet, I think it's fine to leave it as is. However, I'm still a bit worried about performance and massive json strings when you decode them multiple times. 

I have no clue on how to properly solve this issue without applying a transformer.

Reply:
I'm not sure if it's ok or not! but I think that using data transformer will lead us to a strong coupling between two isolated components.

Reply:
Hello guys,
Can I resolve this discussion?

Reply:
editing the constraint object is a bad idea, you the same constraint object will be reused for other validated objects of the same class if any (the metadata belong to the class, not to the instance)